typespec-tests/src/main/java/com/type/union/implementation/models/SendRequest9.java

```diff
@@ -67,13 +67,13 @@ public JsonWriter toJson(JsonWriter jsonWriter) throws IOException {
     @Generated
     public static SendRequest9 fromJson(JsonReader jsonReader) throws IOException {
         return jsonReader.readObject(reader -> {
-            MixedTypesCases prop = null;
+            GetResponseProp4 prop = null;
             while (reader.nextToken() != JsonToken.END_OBJECT) {
                 String fieldName = reader.getFieldName();
                 reader.nextToken();
 
                 if ("prop".equals(fieldName)) {
-                    prop = MixedTypesCases.fromJson(reader);
```

Comment:
ok, order changes again..

Reply:
Mentioned here. The reason is you are renaming the model where TCGC already generate the name.
https://github.com/Azure/autorest.java/pull/2934#discussion_r1732119327

Reply:
ok, I thought the [comment](https://github.com/Azure/autorest.java/pull/2934#discussion_r1732119327) is corresponding to this diff: https://github.com/Azure/autorest.java/pull/2934/files#diff-069b8a8df9585df2f250fe704a3f33872d32991db1db3959d3b943bc9fba2042L181
So the order changes are also due to we renamed?